### PR TITLE
Adding skiplocked support

### PR DESCRIPTION
**Pull request by joshuacullen**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [ ] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Did you follow the commit message conventions explained in [CONTRIBUTING.md](../CONTRIBUTING.md)?

### Description of change

https://github.com/sequelize/sequelize/issues/8910 
Adding SKIP LOCKED to lock queries.

Where should I write tests for this?
Where would you like the docs?

### CODE REVIEW

**Comment by sushantdhiman** [comment]
You can add integration tests in this file https://github.com/sequelize/sequelize/blob/ec0cb5eda8528d1610d8829de56fd1fd13c4f629/test/integration/transaction.test.js#L351

You can add an example of this options

**Comment by joshuacullen** [comment]
Great thanks.

I'll take a look this weekend.

File: test/integration/transaction.test.js [hunk 1]
```diff
@@ -403,6 +403,50 @@ if (current.dialect.supports.transactions) {
...
+              User.create(
+                { username: 'joe'}
+              )
+            ]);
+          }).then(() => {
+            return self.sequelize.transaction().then(t1 => {
+              return User.findAll({
+                limit: 1,
+                lock: true,
+                skipLocked: true,
```
Comment (sushantdhiman) [inline test/integration/transaction.test.js]: In first transaction there is no need to skip locks, I assume all rows are free at this time.

File: test/integration/transaction.test.js [hunk 2]
```diff
@@ -403,6 +403,50 @@ if (current.dialect.supports.transactions) {
...
+                    limit: 1,
+                    lock: true,
+                    skipLocked: true,
+                    transaction: t2
+                  }).then(secondResults => {
+                    expect(secondResults[0].id).to.not.equal(firstUserId);
+                    return Promise.all([
+                      t1.commit(),
+                      t2.commit()
+                    ])
```
Comment (sushantdhiman) [inline test/integration/transaction.test.js]: Missing semi colon failing tests

**Comment by sushantdhiman** [comment]
You need to add a skip lock example [here](https://github.com/sequelize/sequelize/blob/a7e9e2b8ab978230e3f8e39ffdce2a695a102d00/lib/transaction.js#L250)

File: docs/transactions.md [hunk 3]
```diff
@@ -219,3 +219,26 @@ sequelize.transaction({
 The `transaction` option goes with most other options, which are usually the first argument of a method.
 For methods that take values, like `.create`, `.update()`, `.updateAttributes()` etc. `transaction` should be passed to the option in the second argument.
 If unsure, refer to the API documentation for the method you are using to be sure of the signature.
+
+## Locks
```
Comment (sushantdhiman) [inline docs/transactions.md]: Can improve this by adding more examples from [here](https://github.com/sequelize/sequelize/blob/a7e9e2b8ab978230e3f8e39ffdce2a695a102d00/lib/transaction.js#L252-L271)

**Comment by joshuacullen** [comment]
@sushantdhiman Pushed some changes for the above

**Comment by joshuacullen** [comment]
Okay cool, will have a look. Looks like skip locked is only available in MYSQL 8.

I'll remove for mysql and fix up tomorrow.

**Comment by joshuacullen** [comment]
@sushantdhiman Removed support for mysql

**Pull request merged**